lib/jobs/REST-job.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2016, EMC, Inc
+
+'use strict';
+
+var di = require('di');
+
+module.exports = RestJobFactory;
+di.annotate(RestJobFactory, new di.Provide('Job.Rest'));
+di.annotate(RestJobFactory, new di.Inject(
+    'Job.Base', 
+    'Logger', 
+    '_', 
+    'Assert', 
+    'Promise', 
+    'Util', 
+    'JobUtils.HttpTool'));
+
+function RestJobFactory(BaseJob, Logger, _, assert, Promise, util, HttpTool){
+    var logger = Logger.initialize(RestJobFactory);
+    
+    /**
+    * The interface that runs Rest Job from tasks
+    * @constructor
+    */
+    function RestJob(options, context, taskId){
+        var self = this;
+        self.options = options;
+        self.context = context;
+
+        RestJob.super_.call(self, logger, options, context, taskId);
+        self.restClient = new HttpTool();
+    }
+    
+    util.inherits(RestJob, BaseJob);
+    
+    RestJob.prototype._run = function run(){
+        var self = this;
+        logger.info("Runnging A Rest Call");
+        self.restClient.setupRequest(self.options)
+       .catch(function(err){
```

Comment:
The catch will not let your Promise exit earlier,  the afterward `then` will still be executed with resolved `undefined` result. You should first chain `then` next `catch`.